collect_app/src/main/java/org/odk/collect/android/openrosa/OpenRosaResponseParserImpl.kt

```diff
@@ -0,0 +1,220 @@
+                continue
+            }
+
+            val name = xformElement.name
+            if (!name.equals("xform", ignoreCase = true)) {
+                // someone else's extension?
+                continue
+            }
+
+            // this is something we know how to interpret
+            var formId: String? = null
+            var formName: String? = null
+            var version: String? = null
+            var downloadUrl: String? = null
+            var manifestUrl: String? = null
+            var hash: String? = null
+            // don't process descriptionUrl
+
+            val fieldCount = xformElement.childCount
+            for (j in 0 until fieldCount) {
+                if (xformElement.getType(j) != Element.ELEMENT) {
+                    // whitespace
+                    continue
+                }
+
+                val child = xformElement.getElement(j)
+                if (!isXformsListNamespacedElement(child)) {
+                    // someone else's extension?
+                    continue
+                }
+
+                when (child.name) {
+                    "formID" -> {
+                        formId = XFormParser.getXMLText(child, true)
+                        if (formId != null && formId.length == 0) {
+                            formId = null
+                        }
+                    }
+                    "name" -> {
+                        formName = XFormParser.getXMLText(child, true)
+                        if (formName != null && formName.length == 0) {
+                            formName = null
+                        }
+                    }
+                    "version" -> {
+                        version = XFormParser.getXMLText(child, true)
+                        if (version != null && isBlank(version)) {
+                            version = null
+                        }
+                    }
+                    "majorMinorVersion" -> {
+                    }
+                    "descriptionText" -> {
+                    }
+                    "downloadUrl" -> {
+                        downloadUrl = XFormParser.getXMLText(child, true)
+                        if (downloadUrl != null && downloadUrl.length == 0) {
+                            downloadUrl = null
+                        }
+                    }
+                    "manifestUrl" -> {
+                        manifestUrl = XFormParser.getXMLText(child, true)
+                        if (manifestUrl != null && manifestUrl.length == 0) {
+                            manifestUrl = null
+                        }
+                    }
+                    "hash" -> {
+                        hash = XFormParser.getXMLText(child, true)
+                        if (hash != null && (hash.length == 0 || !hash.startsWith("md5:"))) {
+                            hash = null
+                        } else {
+                            hash = hash.substring("md5:".length)
```

Comment:
Since we use this in a few places maybe it's better to have a const for that `md5:`

Reply:
Yeah I'd like to also have extracted an `Md5` data type, but I think I've gone deep enough. I'll do as you suggest and pull out the constant.

Reply:
I'm not sure if putting that constant there is a good idea. It's like our spec that we add that prefix (`md5:`) and not `md5` or `md5-` for example right? So it should be a part of this package and a separate module (in the future) as we agreed on a few minutes later.

Reply:
Ah I was thinking eventually we'd make `Md5` a class and have a `parsePrefixString` method on it or something. I think you're right that I'm jumping ahead there. Right now the prefix is only used in the parser so I'll move the constant there.

Reply:
Ok done!